src/core/dbus.c

```diff
@@ -131,10 +131,15 @@ static int signal_disconnected(sd_bus_message *message, void *userdata, sd_bus_e
         assert(message);
         assert_se(bus = sd_bus_message_get_bus(message));
 
-        if (bus == m->api_bus)
+        if (bus == m->api_bus) {
+                log_debug("Got disconnect on API bus.");
                 bus_done_api(m);
-        if (bus == m->system_bus)
+        }
+        if (bus == m->system_bus) {
                 bus_done_system(m);
+                if (!MANAGER_IS_SYSTEM(m))
+                        log_debug("Got disconnect on system bus.");
```

Comment:
Hmm, above the logging is before `bus_done_api()`, and here it is after `bus_done_system()`. I think it should be before.

Why is the logging excluded when `MANAGER_IS_SYSTEM()`? Maybe add an explanatory comment.

Reply:
If `MANAGER_IS_SYSTEM`, the api bus and the system bus are the same, so we don't need to log about it twice.